### PR TITLE
Circleci

### PR DESCRIPTION
**Pull request by rossatbond**
This PR sets up a CircleCI config for the SDK. All branches will build and be pushed to AWS S3, at 
```
s3://bond-prod-web-assets/sdk/cards/archive/v${FULL_VERSION}-${CIRCLE_BRANCH}-${CIRCLE_SHA1}.js
```
where `FULL_VERSION` is read from the `package.json`. This is a public-read bucket linked to `fastly` and available at `cdn.bond.tech`, for example 

[https://cdn.bond.tech/sdk/cards/archive/v1.0.3-circleci-1061be111fe5fa8420a5647b6b54957c883d1db3.js](https://cdn.bond.tech/sdk/cards/archive/v1.0.3-circleci-1061be111fe5fa8420a5647b6b54957c883d1db3.js)

Storing each build gives us a historical archive and enables testing on branches. 

On `main`, this is done _and_ "formal" releases are written to 
```
s3://bond-prod-web-assets/sdk/cards/v${V1}/bond-sdk-cards.js
s3://bond-prod-web-assets/sdk/cards/v/${V1}/${V2}/${V3}/bond-sdk-cards.js
```
for a semver `V1.V2.V3`. In this scheme, `v${V1}/bond-sdk-cards.js` is always overwritten by a commit to `main` with the same major version but with a higher minor or patch version, while `v/${V1}/${V2}/${V3}/bond-sdk-cards.js` is available and not overwritten unless merges keep the version constant (which I think is against semver).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          at: dist
+      - aws-cli/setup:
+          aws-access-key-id: AWS_ACCESS_KEY_ID
+          aws-secret-access-key: AWS_SECRET_ACCESS_KEY
+          aws-region: AWS_REGION
+          profile-name: default
+      - run: 
+          name: push
+          command: |
+            # get version from code? card or cards?
```
Comment (dominicsardina) [inline .circleci/config.yml:35]: Are these questions we still need to address?

Comment (rossatbond) [reply .circleci/config.yml:35]: Oh good point @dominicsardina . I guess: 
(1) getting the version from `package.json` is safe
(2) `/sdk/card` or `/sdk/cards`?

Comment (dominicsardina) [reply .circleci/config.yml:35]: The repo and npm package are "cards". Am I using them interchangeably somewhere? I can help standardize on "cards" if I used "card" somewhere.

Comment (rossatbond) [reply .circleci/config.yml:35]: All good. Let's use `/sdk/cards`.

**Review by dominicsardina — APPROVED** [review]
In the readme docs, we tell them to use
`<script type="text/javascript" src="https://cdn.bond.tech/sdk/cards/v1/bond-sdk-cards.js"></script>`
Looks like I should add instructions in case they want to also use /v/1/0/3/... ?

**Comment by rossatbond** [comment]
> In the readme docs, we tell them to use
> `<script type="text/javascript" src="https://cdn.bond.tech/sdk/cards/v1/bond-sdk-cards.js"></script>`
> Looks like I should add instructions in case they want to also use /v/1/0/3/... ?

I think that could be an add-on when we want, and see it working, so customers know they can access pinned historical versions there.

**Pull request merged**